custom/icds_reports/utils.py

```diff
@@ -176,3 +184,1196 @@ def render_html(self):
+    image_data = []
+    tmp_image = []
+    img_count = 0
+    count = 1
+    for map_row in map_image_data:
+        lat = map_row['form_location_lat']
+        long = map_row['form_location_long']
+        awc_name = map_row['awc_name']
+        if lat and long:
+            map_data.append({
+                'name': awc_name,
+                'radius': 3,
+                'country': 'IND',
+                'fillKey': 'green',
+                'latitude': lat,
+                'longitude': long
+            })
+        tmp_image.append({'id': count, 'image': 'http://unsplash.it/' + str(300 + count) + '/300'})
+        img_count += 1
+        count += 1
+        if img_count == 4:
+            img_count = 0
+            image_data.append(tmp_image)
+            tmp_image = []
+    if tmp_image:
+        image_data.append(tmp_image)
+
+    return {
+        'map': {
+            'bubbles': map_data,
+            'title': 'PSE Form Submissions',
+            'data': [
+                {
+                    "slug": "pse_forms",
+                    "label": "",
+                    "fills": {
+                        'green': 'green',
+                        'defaultFill': '#9D9D9D',
+                    },
+                    "rightLegend": {
+                    },
+                    "data": None,
+                }
+            ]
+        },
+        'images': image_data
+    }
+
+
+def get_awc_reports_maternal_child(config, month, three_month):
+
+    data = AggChildHealthMonthly.objects.filter(
+        month__range=(datetime(*three_month), datetime(*month)), **config
+    ).values(
+        'month', 'aggregation_level'
+    ).annotate(
+        underweight=(
+            Sum('nutrition_status_moderately_underweight') + Sum('nutrition_status_severely_underweight')
+        ),
+        valid_in_month=Sum('valid_in_month'),
+        immunized=Sum('fully_immunized_on_time') + Sum('fully_immunized_late'),
+        eligible=Sum('fully_immunized_eligible')
+    ).order_by('month')
+
+    prevalence = []
+    immunized = []
+    for row in data:
+        month = row['month']
+        month_in_milliseconds = int(month.strftime("%s")) * 1000
+        prevalence.append([month_in_milliseconds, row['underweight'] / float(row['valid_in_month'] or 1)])
+        immunized.append([month_in_milliseconds, row['immunized'] / float(row['eligible'] or 1)])
+    return {
+        'charts': [
+            [
+                {
+                    'key': 'Prevalence of undernutrition (weight-for-age)',
+                    'values': prevalence,
+                    "classed": "dashed",
+                }
+            ],
+            [
+                {
+                    'key': '% Immunization coverage (at age 1 year)',
+                    'values': immunized,
+                    "classed": "dashed",
+                }
+            ]
+        ]
+    }
+
+
+def get_awc_report_demographics(config, month):
+
+    # map_data = AggAwcMonthly.objects.filter(
```

Comment:
@emord this will be needed in future, will be uncommented when I back to this report.